sphinx/builders/linkcheck.py

```diff
@@ -816,6 +825,7 @@ def setup(app: Sphinx) -> ExtensionMetadata:
     app.add_config_value(
         'linkcheck_report_timeouts_as_broken', False, '', types=frozenset({bool})
     )
+    app.add_config_value('linkcheck_ignore_case', False, '', types=frozenset({bool}))
```

Comment:
NB: along with the documentation, we should also update the config value name here, and the attribute we use to store it on the `HyperlinkAvailabilityCheckWorker` class).